### PR TITLE
Architecture refactoring

### PR DESCRIPTION
**Pull request by hussein-aitlahcen**
This PR will improve the packages by making them easily reusable.
I follow the `nixpkgs` style.

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -1,46 +1,5 @@
-{ pkgs ? import <nixpkgs> {} }:
+with import <nixpkgs> {};
```
Comment (fmenou) [inline default.nix]: I thought it was always a good idea to let the caller pass his own pkgs set, in case he want to override something. Isn't it?

**Review by fmenou — COMMENTED** [review]
Thank you :+1: 

1 comment and I want to test with a real CI build on my end.

File: default.nix (hunk 1 above)
Comment (hussein-aitlahcen) [reply default.nix]: Indeed !

**Review by fmenou — APPROVED** [review]
Good for me, and tested on our CI.

**Pull request merged**